src/nyc_trees/apps/survey/management/commands/send_reminder_emails.py

```diff
@@ -60,25 +66,50 @@ def handle(self, *args, **options):
                                  canceled_at__isnull=True,
                                  expires_at__lte=soon))
 
-        user_reservations = {}
-
-        for reservation in expiring_soon:
-            user_data = user_reservations.get(reservation.user_id, [])
-            user_data.append(reservation)
-            user_reservations[reservation.user_id] = user_data
-
-        if not user_reservations:
+        if not expiring_soon:
             self._log("no emails sent")
             return
 
-        for user_id, reservations in user_reservations.items():
-            try:
-                send_reservation_reminder(user_id, reservations=reservations)
-            except (smtplib.SMTPException, SESError):
-                continue
-            self._log("Email sent to user_id: '%s'" % user_id)
-            ids = map(operator.attrgetter('id'), reservations)
-            (BlockfaceReservation
-             .objects
-             .filter(id__in=ids)
-             .update(reminder_sent_at=now()))
+        users = {}
+        for reservation in expiring_soon:
+            user_data = users.get(reservation.user_id, {})
+            user_reservations = user_data.get(_RESERVATIONS, [])
+            user_reservations.append(reservation)
+            user_data[_RESERVATIONS] = user_reservations
+            user_data[_REMAINING_RETRIES] = _RETRY_COUNT
+            users[reservation.user_id] = user_data
+
+        remaining_retries = True
+        while remaining_retries:
+            remaining_retries = False
+            for user_id, user_data in users.items():
+                user_remaining_retries = user_data[_REMAINING_RETRIES]
+
+                if user_remaining_retries == 0:
+                    continue
+
+                reservations = user_data[_RESERVATIONS]
+
+                try:
+                    send_reservation_reminder(
+                        user_id, reservations=reservations)
+                except (smtplib.SMTPException, SESError):
+                    user_remaining_retries -= 1
+                    user_data[_REMAINING_RETRIES] = user_remaining_retries
+                    if user_remaining_retries > 0:
+                        remaining_retries = True
+                    continue
+                finally:
+                    # sleep to avoid hitting request-per-second limits in AWS
+                    # given that we have plenty of time for this tasks to run
+                    time.sleep(0.5)
```

Comment:
Good catch. I had previously not read the doc that states in addition to the 50k/24h limit there is a 14/second limit.

It would be helpful to drop a link to http://docs.aws.amazon.com/ses/latest/DeveloperGuide/limits.html in the comment